src/Kafka/Avro/SchemaRegistry.hs

```diff
@@ -135,18 +175,45 @@ loadSubjectSchema sr (Subject sbj) (Version version) = do
                      Error e   -> Left e
 
 
+-- | Get the schema ID.
+-- If the schema registry is configured to auto-register schemas,
+-- this posts the schema to the schema to the schema registry.
+-- Otherwise, this searches for a compatible schema and returns a 'SchemaRegistryNoCompatibleSchemaFound'
+-- if none is found.
 sendSchema :: MonadIO m => SchemaRegistry -> Subject -> Schema -> m (Either SchemaRegistryError SchemaId)
 sendSchema sr subj sc = do
+  let schemaName = fullTypeName sc
   sid <- cachedId sr subj schemaName
   case sid of
     Just sid' -> return (Right sid')
-    Nothing   -> do
-      res <- liftIO $ putSchema sr subj (RegisteredSchema sc)
-      traverse_ (cacheId sr subj schemaName) res
-      traverse_ (\sid' -> cacheSchema sr sid' sc) res
-      pure res
-  where
-    schemaName = fullTypeName sc
+    Nothing -> if cAutoRegisterSchemas (srConfig sr)
+                then registerSchema sr subj sc
+                else getCompatibleSchema sr subj sc
```

Comment:
The name of this function isn't the best for what it does anymore,
but I guess renaming it would be a breaking change.